pkg/shed/db.go

```diff
@@ -50,16 +56,22 @@ type DB struct {
 // NewDB constructs a new DB and validates the schema
 // if it exists in database on the given path.
 // metricsPrefix is used for metrics collection for the given DB.
-func NewDB(path string) (db *DB, err error) {
+func NewDB(path string, o *Options) (db *DB, err error) {
+	if o == nil {
+		o = &Options{
+			OpenFilesLimit:     defaultOpenFilesLimit,
+			BlockCacheCapacity: defaultBlockCacheCapacity,
+			WriteBufferSize:    defaultWriteBufferSize,
+		}
+	}
 	var ldb *leveldb.DB
 	if path == "" {
 		ldb, err = leveldb.Open(storage.NewMemStorage(), nil)
 	} else {
 		ldb, err = leveldb.OpenFile(path, &opt.Options{
-			OpenFilesCacheCapacity: openFileLimit,
-			BlockCacheCapacity:     blockCacheCapacity,
-			DisableSeeksCompaction: true,
-			WriteBuffer:            writeBuffer,
+			OpenFilesCacheCapacity: int(o.OpenFilesLimit),
+			BlockCacheCapacity:     int(o.BlockCacheCapacity),
+			WriteBuffer:            int(o.WriteBufferSize),
```

Comment:
This PR removes DisableSeeksCompaction option. If there are some noticeable additional i/o, one suggestion is to try to put it back.

Reply:
Indeed. This is intentional. Perhaps maybe best it to actually add another CLI flag for this... I'll do this right away